sqlalchemy-stubs/sql/sqltypes.pyi

```diff
@@ -327,7 +329,7 @@ class ARRAY(SchemaEventTarget, Indexable, Concatenable, TypeEngine[List]):
 class TupleType(TypeEngine[TupleType]):
     types: Any = ...
     def __init__(self, *types: Any) -> None: ...
-    def result_processor(self, dialect: Any, coltype: Any) -> None: ...
+    def result_processor(self, dialect: Any, coltype: Any) -> NoReturn: ...
```

Comment:
this is interesting.  result_processor() overall returns a callable function, as you've done in type_api.pyi below.   if a subclass happens to have NotImplemetnedError as its implementation, does that mean result_processor() on that subtype changes the type signature?  I would think for the stubs, result_processor() should only be on TypeEngine since that's the signature.

Reply:
>  I would think for the stubs, result_processor() should only be on TypeEngine since that's the signature.

We can certainly do that. I was just reflecting the codebase. This technically changes the type signature, but `NoReturn` is treated like `Any`, so type checkers ignore it.

Reply:
OK I would just remove the whole thing for now because this is not the "published" API of that method, like maybe TupleType would implement it someday.